x-pack/plugins/ml/public/application/explorer/swimlane_pagination.tsx

```diff
@@ -51,25 +51,33 @@ export const SwimLanePagination: FC<SwimLanePaginationProps> = ({
 
   const pageCount = Math.ceil(cardinality / perPage);
 
-  const items = [5, 10, 20, 50, 100].map((v) => {
-    return (
-      <EuiContextMenuItem
-        key={`${v}_rows`}
-        icon={v === perPage ? 'check' : 'empty'}
-        onClick={() => {
-          closePopover();
-          setPerPage(v);
-        }}
-        data-test-subj={`${v} rows`}
-      >
-        <FormattedMessage
-          id="xpack.ml.explorer.swimLaneSelectRowsPerPage"
-          defaultMessage="{rowsCount} rows"
-          values={{ rowsCount: v }}
-        />
-      </EuiContextMenuItem>
-    );
-  });
+  const rowOptions = [5, 10, 20, 50, 100];
+  const items = rowOptions.reduce((acc, v) => {
+    if (v <= cardinality) {
+      acc.push(v);
+    } else if (acc.length === 0 || acc[acc.length - 1] < cardinality) {
+      acc.push(v);
+    }
+    return acc;
+  }, [] as number[]);
```

Comment:
Nit: You could move this to a function and add some jest unit tests for it.

Reply:
Moved function to a single file & added unit tests in: [#3035a1d](https://github.com/elastic/kibana/pull/194531/commits/3035a1da88282892b9de40f0843ed56dc51596c5)